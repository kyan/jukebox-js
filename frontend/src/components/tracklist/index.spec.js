import React from 'react'
import { shallow } from 'enzyme'
import Tracklist from './index'
import MockTrackListJson from '../../__mockData__/api'

describe('Tracklist', () => {
  let wrapper
  let tracks = MockTrackListJson()
  const images = {
    'spotify:album:5OVGwMCexoHavOar6v4al5': 'album-image.jpg'
  }

  describe('render', () => {
    Date.now = jest.fn(() => { return 8208000 })

    wrapper = shallow(
      <Tracklist
        tracks={tracks.map(item => item.track)}
        images={images}
        currentTrack={tracks[0].track}
      />
    )

    it('renders the as expected', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('when no track', () => {
    it('renders nothing', () => {
      wrapper = shallow(
        <Tracklist />
      )

      expect(wrapper.instance()).toBeNull()
    })
  })
})
