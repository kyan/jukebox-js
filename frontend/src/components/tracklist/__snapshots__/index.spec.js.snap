// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tracklist render when disabled renders the as expected 1`] = `
<List
  relaxed={true}
>
  <ListItem
    key="0-spotify:track:1yzSSn5Sj1azuo7RgwvDb3"
  >
    <span
      className="image current-track"
    >
      <Image
        as="img"
        inline={true}
        size="tiny"
        src="album-image.jpg"
        ui={true}
      />
    </span>
    <ListContent
      className=""
    >
      <ListHeader
        as="h4"
      >
        No Time for Caution
      </ListHeader>
      <ListDescription>
        Hans Zimmer
         
        <small>
          (
          4:06
          )
        </small>
      </ListDescription>
    </ListContent>
  </ListItem>
  <ListItem
    key="1-local:track:Soundtracks/Silent%20Running%20OST/Silent%20Running%20"
  >
    <span
      className="image current-track"
    >
      <Image
        as="img"
        inline={true}
        size="tiny"
        src="default-artwork.png"
        ui={true}
      />
    </span>
    <ListContent
      className=""
    >
      <ListHeader
        as="h5"
      >
        2:18
      </ListHeader>
      <ListHeader
        as="h4"
      >
        Silent Running
      </ListHeader>
      <ListDescription>
        Joan Baez
         
        <small>
          (
          2:04
          )
        </small>
      </ListDescription>
    </ListContent>
  </ListItem>
</List>
`;

exports[`Tracklist render when enabled renders the as expected 1`] = `
<List
  relaxed={true}
>
  <ListItem
    key="0-spotify:track:1yzSSn5Sj1azuo7RgwvDb3"
  >
    <span
      className="image current-track"
    >
      <Image
        as="img"
        inline={true}
        size="tiny"
        src="album-image.jpg"
        ui={true}
      />
    </span>
    <ListContent
      className=""
    >
      <ListHeader
        as="h4"
      >
        No Time for Caution
      </ListHeader>
      <ListDescription>
        Hans Zimmer
         
        <small>
          (
          4:06
          )
        </small>
      </ListDescription>
    </ListContent>
  </ListItem>
  <ListItem
    key="1-local:track:Soundtracks/Silent%20Running%20OST/Silent%20Running%20"
  >
    <a
      className="image"
      onClick={[Function]}
    >
      <Reveal
        animated="fade"
      >
        <RevealContent
          visible={true}
        >
          <Image
            as="img"
            inline={true}
            src="default-artwork.png"
            ui={true}
          />
        </RevealContent>
        <RevealContent
          hidden={true}
        >
          <Icon
            as="i"
            bordered={true}
            color="red"
            name="close"
          />
        </RevealContent>
      </Reveal>
    </a>
    <ListContent
      className=""
    >
      <ListHeader
        as="h5"
      >
        2:18
      </ListHeader>
      <ListHeader
        as="h4"
      >
        Silent Running
      </ListHeader>
      <ListDescription>
        Joan Baez
         
        <small>
          (
          2:04
          )
        </small>
      </ListDescription>
    </ListContent>
  </ListItem>
</List>
`;

exports[`Tracklist tracklist but nothing cued up does not mark anything as current 1`] = `
<List
  relaxed={true}
>
  <ListItem
    key="0-spotify:track:1yzSSn5Sj1azuo7RgwvDb3"
  >
    <a
      className="image"
      onClick={[Function]}
    >
      <Reveal
        animated="fade"
      >
        <RevealContent
          visible={true}
        >
          <Image
            as="img"
            inline={true}
            src="default-artwork.png"
            ui={true}
          />
        </RevealContent>
        <RevealContent
          hidden={true}
        >
          <Icon
            as="i"
            bordered={true}
            color="red"
            name="close"
          />
        </RevealContent>
      </Reveal>
    </a>
    <ListContent
      className="track-info"
    >
      <ListHeader
        as="h4"
      >
        No Time for Caution
      </ListHeader>
      <ListDescription>
        Hans Zimmer
         
        <small>
          (
          4:06
          )
        </small>
      </ListDescription>
    </ListContent>
  </ListItem>
  <ListItem
    key="1-local:track:Soundtracks/Silent%20Running%20OST/Silent%20Running%20"
  >
    <a
      className="image"
      onClick={[Function]}
    >
      <Reveal
        animated="fade"
      >
        <RevealContent
          visible={true}
        >
          <Image
            as="img"
            inline={true}
            src="default-artwork.png"
            ui={true}
          />
        </RevealContent>
        <RevealContent
          hidden={true}
        >
          <Icon
            as="i"
            bordered={true}
            color="red"
            name="close"
          />
        </RevealContent>
      </Reveal>
    </a>
    <ListContent
      className="track-info"
    >
      <ListHeader
        as="h4"
      >
        Silent Running
      </ListHeader>
      <ListDescription>
        Joan Baez
         
        <small>
          (
          2:04
          )
        </small>
      </ListDescription>
    </ListContent>
  </ListItem>
</List>
`;
